templates/jenkins-config.xml

```diff
@@ -9,4 +9,5 @@
     <disableSignup>true</disableSignup>
     <enableCaptcha>false</enableCaptcha>
   </securityRealm>
+  <slaveAgentPort>{{jnlp_port}}</slaveAgentPort>
```

Comment:
So say that I run:

```
juju set jenkins jnlp-port=""
```

(or deploy with a config.yaml set likewise) with the intent of using a random port. Will the XML above do the right thing? I.e. allocate a random port of the <slaveAgentPort> node has empty text.

Reply:
That's a great point. It's set as an integer in config.yaml so I think Juju may reject an empty value.

According to the docs, you can set it to a random port via -1, or to 0 to disable the feature entirely (search inline for "jenkins.model.Jenkins.slaveAgentPort"):

https://wiki.jenkins.io/display/JENKINS/Features+controlled+by+system+properties

I'll push a fix documenting that, and I guess I should add logic to ignore any invalid value (i.e. anything not either -1, 0 or an integer between 1 and 65535) as the default.

Reply:
Thanks, it looks good to me. I have just a point about unit tests left.